src/Appwrite/Platform/Tasks/ScheduleExecutions.php

```diff
@@ -52,42 +50,54 @@ protected function enqueueResources(Group $pools, Database $dbForPlatform, calla
             }
 
             $scheduledAt = new \DateTime($schedule['schedule']);
-            if ($scheduledAt <= $intervalEnd) {
-                continue;
-            }
 
             $data = $dbForPlatform->getDocument(
                 'schedules',
                 $schedule['$id'],
             )->getAttribute('data', []);
 
             $delay = $scheduledAt->getTimestamp() - (new \DateTime())->getTimestamp();
+            $delay = max($delay, 0);
 
             $this->updateProjectAccess($schedule['project'], $dbForPlatform);
 
-            \go(function () use ($queueForFunctions, $schedule, $scheduledAt, $delay, $data) {
+            \go(function () use ($queueForFunctions, $schedule, $scheduledAt, $delay, $data, $dbForPlatform) {
                 Co::sleep($delay);
 
-                $queueForFunctions->setType('schedule')
+                $executedAt = new \DateTime();
+                $actualDelay = $executedAt->getTimestamp() - $scheduledAt->getTimestamp();
```

Comment:
Let's rename actualDelay to something else. We might want to call them relativeDelay and absoluteDelay. relative for coroutines, absolute for execution.
Or coroutineDelay and executionDelay